### PR TITLE
logging: add error to udp logs

### PR DESCRIPTION
**Pull request by jzelinskie**
This also makes the format of UDP and HTTP logs a little more similar.

### CODE REVIEW

File: udp/udp.go [hunk 1]
```diff
@@ -68,19 +68,22 @@ func (s *Server) serve(listenAddr string) error {
...
 		}
 
-		start := time.Now()
-
 		go func() {
-			response, action := s.handlePacket(buffer[:n], addr)
-			pool.GiveSlice(buffer)
+			start := time.Now()
+			response, action, err := s.handlePacket(buffer[:n], addr)
+			defer pool.GiveSlice(buffer)
```
Comment (pushrax) [inline udp/udp.go]: Any particular reason to defer this?

File: udp/protocol.go [hunk 2]
```diff
@@ -96,8 +102,9 @@ func (s *Server) handlePacket(packet []byte, addr *net.UDPAddr) (response []byte
 
 	case announceActionID:
 		actionName = "announce"
-		ann, err := s.newAnnounce(packet, addr.IP)
 
+		var ann *models.Announce
+		ann, err = s.newAnnounce(packet, addr.IP)
```
Comment (pushrax) [inline udp/protocol.go]: There needs to be a better way of doing this in Go

File: udp/udp.go (hunk 1 above)
Comment (jzelinskie) [inline udp/udp.go]: I didn't really want it to count in the timing, but wanted to keep it in the context of the `handlePacket` call.

File: udp/protocol.go (hunk 2 above)
Comment (jzelinskie) [inline udp/protocol.go]: Let me know when you figure it out :\

**Pull request merged**